python-modules/cis_change_service/cis_change_service/idp.py

```diff
@@ -0,0 +1,131 @@
+import json
+import logging
+
+from functools import wraps
+from flask import request
+from flask import _request_ctx_stack
+from six.moves.urllib.request import urlopen
+from jose import jwt
+
+from cis_change_service import get_config
+from cis_change_service.exceptions import AuthError
+
+logger = logging.getLogger(__name__)
+
+CONFIG = get_config()
+
+AUTH0_DOMAIN = CONFIG('auth0_domain', namespace='change_service', default='auth-dev.mozilla.auth0.com')
+API_IDENTIFIER = CONFIG('api_identifier', namespace='change_service', default='https://change.sso.allizom.org')
+ALGORITHMS = CONFIG('algorithms', namespace='change_service', default='RS256')
+
+
+# Format error response and append status code
+def get_token_auth_header():
+    """Obtains the Access Token from the Authorization Header
+    """
+    auth = request.headers.get("Authorization", None)
+    if not auth:
+        raise AuthError({"code": "authorization_header_missing",
+                        "description":
+                            "Authorization header is expected"}, 401)
+
+    parts = auth.split()
+
+    if parts[0].lower() != "bearer":
+        raise AuthError({"code": "invalid_header",
+                        "description":
+                            "Authorization header must start with"
+                            " Bearer"}, 401)
+    elif len(parts) == 1:
+        raise AuthError({"code": "invalid_header",
+                        "description": "Token not found"}, 401)
+    elif len(parts) > 2:
+        raise AuthError({"code": "invalid_header",
+                        "description":
+                            "Authorization header must be"
+                            " Bearer token"}, 401)
+
+    token = parts[1]
+    return token
+
+
+def get_jwks():
+    # XXX TBD do this with request purely instead of six
+    jsonurl = urlopen("https://" + AUTH0_DOMAIN + "/.well-known/jwks.json")
+    jwks = json.loads(jsonurl.read())
+    return jwks
+
+
+def requires_auth(f):
+    """Determines if the Access Token is valid
+    """
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        if CONFIG(
```

Comment:
hard to read

Reply:
Do you mean because it's a decorator?  Or ...

Reply:
no, its the syntax that is a little odd compared to the rest of the file because it breaks line several times with comparisons (`==`) in the middle, wasn't used to that
ie style stuff

Reply:
You're totally right this was hard to reason about.  I fixed it.